src/rules/extend-require-placeholder/index.js

```diff
@@ -0,0 +1,28 @@
+import { utils } from "stylelint"
+
+export const ruleName = "extend-require-placeholder"
+
+export const messages = utils.ruleMessages(ruleName, {
+  rejected: "Use a placeholder selector (e.g. %some-placeholder) with @extend",
+})
+
+export default function (actual) {
+  return function (root, result) {
+    const validOptions = utils.validateOptions(result, ruleName, { actual })
+    if (!validOptions) { return }
+
+    root.walkAtRules("extend", atrule => {
+      const isPlaceholder = (/^%/).test(atrule.params.trim())
```

Comment:
This would probably be some degree faster if you just checked the first character instead of running a regexp (`atrule.params.trim()[0] === '%'`)